arch/arc/core/mpu/arc_mpu.c

```diff
@@ -140,6 +143,127 @@ static inline s32_t _mpu_probe(u32_t addr)
 }
 #endif
 
+/**
+ * This internal function is utilized by the MPU driver to parse the intent
+ * type (i.e. THREAD_STACK_REGION) and return the correct region index.
+ */
+static inline u32_t _get_region_index_by_type(u32_t type)
+{
+	/*
+	 * The new MPU regions are allocated per type after the statically
+	 * configured regions. The type is one-indexed rather than
```

Comment:
I think MPUv2 and MPUv3 use a different layout/allocation for the indexes, please update comments. It would help to add a short description of the layout, as well.

Reply:
added